test/firebase_auth_mocks_test.dart

```diff
@@ -245,6 +322,56 @@ void main() {
         throwsA(isA<FirebaseAuthException>()),
       );
     });
+
+    test('sendSignInLinkToEmail', () async {
+      final auth = MockFirebaseAuth(
+        authExceptions: AuthExceptions(
+          sendSignInLinkToEmail: FirebaseAuthException(code: 'invalid-email'),
+        ),
+      );
+
+      expect(
+        () async => await auth.sendSignInLinkToEmail(
+          email: 'test@example.com',
+          actionCodeSettings: ActionCodeSettings(
+            url: 'https://example.com',
+            handleCodeInApp: true,
+          ),
+        ),
+        throwsA(isA<FirebaseAuthException>()),
+      );
+    });
+
+    test('confirmPasswordReset', () async {
+      final auth = MockFirebaseAuth(
+        authExceptions: AuthExceptions(
+          confirmPasswordReset:
+              FirebaseAuthException(code: 'invalid-action-code'),
+        ),
+      );
+
+      expect(
+        () async => await auth.confirmPasswordReset(
+          code: 'code',
+          newPassword: 'password',
+        ),
+        throwsA(isA<FirebaseAuthException>()),
+      );
+    });
+
+    test('verifyPasswordResetCode', () async {
+      final auth = MockFirebaseAuth(
+        authExceptions: AuthExceptions(
+          verifyPasswordResetCode:
+              FirebaseAuthException(code: 'invalid-action-code'),
+        ),
+      );
+
+      expect(
+        () async => await auth.verifyPasswordResetCode('code'),
```

Comment:
I believe you could write `() => auth.verifyPasswordResetCode('code')` for the same result, here and above. But win that case, we might have to use `completion` instead of `returnsNormally`.

https://api.flutter.dev/flutter/package-test_api_expect/expect.html
> Certain matchers, like [completion](https://api.flutter.dev/flutter/package-test_api_expect/completion.html) and [throwsA](https://api.flutter.dev/flutter/package-test_api_expect/throwsA.html), either match or fail asynchronously. When you use [expect](https://api.flutter.dev/flutter/package-test_api_expect/expect.html) with these matchers, it ensures that the test doesn't complete until the matcher has either matched or failed. If you want to wait for the matcher to complete before continuing the test, you can call [expectLater](https://api.flutter.dev/flutter/package-test_api_expect/expectLater.html) instead and await the result.

Reply:
There are so many matchers, I think I have never seen ```completion``` before. I will check it out next time though!